packages/quantic/force-app/main/default/lwc/quanticRecommendationList/quanticRecommendationList.js

```diff
@@ -0,0 +1,199 @@
+ * The `QuanticRecommendationList` component displays recommendations by applying one or more result templates in different layouts.
+ * @fires CustomEvent#registerrecommendationtemplates
+ * @category Recommendation
+ * @example
+ * <c-quantic-recommendation-list engine-id={engineId} recommendation={recommendationId} fields-to-include="objecttype,gdfiletitle" number-of-recommendations=”3” number-of-recommendations-per-row=”10” heading-level="1"></c-quantic-recommendation-list>
+ */
+export default class QuanticRecommendationList extends LightningElement {
+  /**
+   * The ID of the engine instance the component registers to.
+   * @api
+   * @type {string}
+   */
+  @api engineId;
+  /**
+   * The recommendation identifier used by the Coveo platform to retrieve recommended documents.
+   * @api
+   * @type {string}
+   * @default {'recommendation'}
+   */
+  @api recommendation = 'recommendation';
+  /**
+   * The total number of recommendations to fetch.
+   * @api
+   * @type {number}
+   * @default {10}
+   */
+  @api numberOfRecommendations = 10;
+  /**
+   * The number of recommendations to display, per row.
+   * Each recommendation in the row will be displayed as
+   * 1/numberOfRecommendationsPerRow of the container width.
+   * @api
+   * @type {number}
+   * @default {3}
+   */
+  @api numberOfRecommendationsPerRow = 3;
+  /**
+   * A list of fields to include in the query results, separated by commas.
+   * @api
+   * @type {string}
+   * @defaultValue `'date,author,source,language,filetype,parents,sfknowledgearticleid,sfid,sfkbid,sfkavid'`
+   */
+  @api fieldsToInclude =
+    'date,author,source,language,filetype,parents,sfknowledgearticleid,sfid,sfkbid,sfkavid';
+  /**
+   * The label of the component. This label is displayed in the component header.
+   * @api
+   * @type {string}
+   */
+  @api label;
+  /**
+   * The Heading level to use for the heading label, accepted values are integers from 1 to 6.
+   * @type {number}
+   * @default {1}
+   */
+  @api headingLevel = 1;
+
+  /** @type {RecommendationListState} */
+  @track state;
+  /** @type {RecommendationList} */
+  recommendationList;
+  /** @type {boolean} */
+  showPlaceholder = false;
+  /** @type {Function} */
+  unsubscribe;
+  /** @type {ResultTemplatesManager} */
+  resultTemplatesManager;
+  /** @type {CoveoHeadlessRecommendation} */
+  headless;
+  /** @type {boolean} */
+  hasInitializationError = false;
+
+  connectedCallback() {
+    registerComponentForInit(this, this.engineId);
+    this.setRecommendationWidth();
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize);
+  }
+
+  /**
+   * @param {RecommendationEngine} engine
+   */
+  initialize = (engine) => {
+    this.headless = getHeadlessBundle(this.engineId);
+    this.recommendationList = this.headless.buildRecommendationList(engine, {
+      options: {
+        id: this.recommendation,
+        numberOfRecommendations: Number(this.numberOfRecommendations),
```

Comment:
yep, otherwise the Headless controller will crash.

Reply:
No :) I meant, calling `Numer(something)` sadly doesn't guarantee a number haha.

`Number('foo') => NaN`
`Number(undefined) => NaN`

etc.

Reply:
Ah gotcha now. Actually we don't need to do that cause a validation is done in Headless and since we implemented the logic to handle component initialization error so in the case where for example an end user introduces the value `foo` the Quantic Component error will be displayed in addition to the lightning modal that will notify the user of the issue and a detailed message will be logged saying that the property `numberOfRecommendations` needs to be a valid number.